src/main/java/edu/ksu/canvas/impl/ExternalToolImpl.java

```diff
@@ -0,0 +1,157 @@
+        JsonObject toolJson = gson.toJsonTree(tool).getAsJsonObject();
+        Response response = canvasMessenger.sendJsonPostToCanvas(oauthToken, url, toolJson);
+        return responseParser.parseToObject(ExternalTool.class, response);
+    }
+
+    @Override
+    public Optional<ExternalTool> editExternalToolInCourse(String courseId, ExternalTool tool) throws IOException {
+        LOG.debug("Editing external tool \"" + tool.getName() + "\" in course " + courseId);
+        String url = buildCanvasUrl("courses/" + courseId + "/external_tools/" + tool.getId(), Collections.emptyMap());
+        return editExternalTool(url, tool);
+    }
+
+    @Override
+    public Optional<ExternalTool> editExternalToolInAccount(String accountId, ExternalTool tool) throws IOException {
+        LOG.debug("Editing external tool \"" + tool.getName() + "\" in course " + accountId);
+        String url = buildCanvasUrl("accounts/" + accountId + "/external_tools/" + tool.getId(), Collections.emptyMap());
+        return editExternalTool(url, tool);
+    }
+
+    private Optional<ExternalTool> editExternalTool(String url, ExternalTool tool) throws IOException {
+        if(tool.getId() == null) {
+            throw new IllegalArgumentException("Tool being edited must have a tool ID");
+        }
+        Gson gson = GsonResponseParser.getDefaultGsonParser();
+        JsonObject toolJson = gson.toJsonTree(tool).getAsJsonObject();
+        Response response = canvasMessenger.sendJsonPutToCanvas(oauthToken, url, toolJson);
+        return responseParser.parseToObject(ExternalTool.class, response);
+    }
+
+    public Optional<ExternalTool> deleteExternalToolInCourse(String courseId, Integer toolId) throws IOException {
+        LOG.debug("Deleting external tool " + toolId + " from course " + courseId);
+        String url = buildCanvasUrl("courses/" + courseId + "/external_tools/" + toolId, Collections.emptyMap());
+        Response response = canvasMessenger.deleteFromCanvas(oauthToken, url, Collections.emptyMap());
+        return responseParser.parseToObject(ExternalTool.class, response);
+    }
+
+    public Optional<ExternalTool> deleteExternalToolInAccount(String accountId, Integer toolId) throws IOException {
+        LOG.debug("Deleting external tool " + toolId + " from account " + accountId);
+        String url = buildCanvasUrl("accounts/" + accountId + "/external_tools/" + toolId, Collections.emptyMap());
+        Response response = canvasMessenger.deleteFromCanvas(oauthToken, url, Collections.emptyMap());
+        return responseParser.parseToObject(ExternalTool.class, response);
+    }
+
+    /**
+     * Ensure that a tool object is valid for creation. The API requires certain fields to be filled out.
+     * Throws an IllegalArgumentException if the conditions are not met.
+     * @param tool The external tool object we are trying to create
+     */
+    private void validateToolForCreation(ExternalTool tool) {
```

Comment:
The naming of this method implied it returns a boolean (to me anyway). Not sure if it needs a change.